src/graphics/texture.rs

```diff
@@ -147,45 +173,7 @@ impl Texture {
 
     /// Upload texture to GPU with given TextureParams
     pub fn from_data_and_format(ctx: &mut Context, bytes: &[u8], params: TextureParams) -> Texture {
-        assert_eq!(
-            params.format.size(params.width, params.height),
-            bytes.len() as u32
-        );
-
-        let (internal_format, format, pixel_type) = params.format.into();
-
-        unsafe {
-            ctx.cache.store_texture_binding(0);
-
-            let mut texture: GLuint = 0;
-            glGenTextures(1, &mut texture as *mut _);
-            ctx.cache.bind_texture(0, texture);
```

Comment:
just for reference: this is the place where glActiveTexture() was missing - not sure if by accident or intentional